docs/lib/build.js

```diff
@@ -0,0 +1,132 @@
+const { join, dirname, basename, extname, sep, posix } = require('path')
+const fs = require('@npmcli/fs')
+const ignoreWalk = require('ignore-walk')
+const yaml = require('yaml')
+const parseFrontMatter = require('front-matter')
+const mdx = require('@mdx-js/mdx')
+
+const checkNav = require('./check-nav.js')
+const transform = require('./transform.js')
+
+const DOC_EXT = '.md'
+
+const mkDirs = async (paths) => {
+  const uniqDirs = [...new Set(paths.map((p) => dirname(p)))]
+  return Promise.all(uniqDirs.map((d) => fs.mkdir(d, { recursive: true })))
+}
+
+const rmAll = (...dirs) => Promise.all(dirs.map((d) => fs.rm(d, { recursive: true, force: true })))
+const readDocs = (path) => ignoreWalk({ path }).then(ps => ps.filter(p => extname(p) === DOC_EXT))
+const readMd = (path) => fs.readFile(path, 'utf-8').then(parseFrontMatter)
+const readHtml = (path) => fs.readFile(path, 'utf-8')
+const readYaml = (path) => fs.readFile(path, 'utf-8').then(yaml.parse)
+const makeTransforms = (...args) => (src, trs) => trs.reduce((acc, tr) => tr(acc, ...args), src)
+
+const run = async ({ content, verify, template, nav, man, html, md }) => {
+  const [contentPaths, templateFile, navFile] = await Promise.all([
+    readDocs(content),
+    readHtml(template),
+    readYaml(nav),
+    rmAll(man, html, md),
```

Comment:
So this always removed all the docs every time, meaning that as long as we are running this the docs will be guaranteed to be built fresh? This solves an existing problem where `make` would sometimes not rebuild docs cause of timestamp shenanigans.

Reply:
Yes it does everything from scratch each time. My only concern with this was how long it would take, so I profiled it with and without running each doc through `mdx`. It was ~3 seconds without doing mdx verification and ~5 seconds with it.